From b88fd0c77913b46d89c13e6c0c8c7460d35d176b Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Mon, 12 Aug 2013 15:11:57 +0300
Subject: [PORT FROM MAIN] Telephony: Unregister for sim ready in dispose method

BZ: 127197

Constructor of CatService uses registerForReady() but
unregisterForReady() is not called in dispose method.

Event registrations should be unregistered in dispose
method. So, unregister for sim ready in dispose method
to avoid any issues.

Change-Id: Ia06043d3c1b25e9bacac6514cab1b03f4be2f96d
Orig-Change-Id: I9565fdf124915de2109e7d7f9c002659d6f1cf66
Category: aosp improvement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
---
 .../android/internal/telephony/cat/CatService.java |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/cat/CatService.java b/src/java/com/android/internal/telephony/cat/CatService.java
index c8243ce..7ad7015 100644
--- a/src/java/com/android/internal/telephony/cat/CatService.java
+++ b/src/java/com/android/internal/telephony/cat/CatService.java
@@ -162,6 +162,10 @@ public class CatService extends Handler implements AppInterface {
         mCmdIf.unSetOnCatEvent(this);
         mCmdIf.unSetOnCatCallSetUp(this);
 
+        if (mUiccApplication != null) {
+            mUiccApplication.unregisterForReady(this);
+        }
+
         removeCallbacksAndMessages(null);
         PhoneFactory.getDefaultPhone().unregisterForServiceStateChanged(this);
     }
-- 
1.7.4.1

